### PR TITLE
[buteo-sync-plugins-social] Optimise FB Image sync selection. Contributes to MER#1124

### PR DESCRIPTION
**Pull request by chriadam**
This commit ensures that we select image and thumbnail URLs based on
the physical resolution of the device screen.

Contributes to MER#1124

### CODE REVIEW

**Comment by chriadam** [comment]
@saukko @jpetrell @antseppa

File: src/facebook/facebook-images/facebookimagesyncadaptor.cpp [hunk 1]
```diff
@@ -308,29 +308,48 @@ void FacebookImageSyncAdaptor::imagesFinishedHandler()
...
+                                        static_cast<int>(image.value(QLatin1String("height")).toDouble()),
+                                        image.value(QLatin1String("source")).toString());
+        }
+
+        int optimalThumbnailWidth = 360; // if we could access QScreen: QGuiApplication::primaryScreen()->size().width() / 3;
+        int optimalImageWidth = 1536;    // if we could access QScreen: QGuiApplication::primaryScreen()->size().width();
+        bool foundOptimalThumbnail = false, foundOptimalImage = false;
+        std::sort(imageSources.begin(), imageSources.end());
+        Q_FOREACH (const ImageSource &img, imageSources) {
+            if (!foundOptimalThumbnail && qMin(img.width, img.height) >= optimalThumbnailWidth) {
```
Comment (chriadam) [inline src/facebook/facebook-images/facebookimagesyncadaptor.cpp]: I'm not 100% sure about the qMin() here, but if we don't do it then we could end up downloading lower-resolution images than would be ideal in the case where the image is landscape oriented.  Note that with this check, we pretty much hit the "fall back to the largest" image every single time, due to the resolutions which FB supports.

**Comment by chriadam** [comment]
Updated, requires https://github.com/nemomobile/lipstick/pull/357

**Comment by antseppa** [comment]
LGTM (only tried on a phone, unfortunately wasn't able to test on tablet at this moment).

**Comment by chriadam** [comment]
@jpetrell can you also look at this one and the lipstick PR it depends on?

**Pull request merged**